### ISSUE TITLE
Using old API endpoint to fetch gifs.

### ISSUE THREAD

**Issue by oooShiny** [gfycat/gfycat.js#25]
The call to get gifs is using an endpoint that seems to have been retired:

`loadJSONP("https://gfycat.com/cajax/get/" + gfyId, function (data) {` 

The cajax/get/ endpoint returns a 404 as of recently.

[Here's the line of code](https://github.com/gfycat/gfycat.js/blob/584e1f977f82fb69bed6962bfe55093fe8a9d591/web/js/gfyObject.js#L369).

**Comment by TPME** [comment]
Do you happen to know the new endpoint, so we can modify the script to make it work again?

Alternatively: How do we solve this differently?

**Comment by oooShiny** [comment]
It looks like the new endpoint might be `https://api.gfycat.com/v1/gfycats/{gfyid}` according to the [API docs](https://developers.gfycat.com/api/#getting-gfycats). I created a PR that swaps out the old endpoint with this new one, but haven't been able to do any testing on it to see if everything still works.

**Comment by TPME** [comment]
I've found that new endpoint as well. It does not support JSONP anymore though which is a shame because it's required for this library!

I've made my own solution for the problem by proxying the API call to gfycat through my own servers and then adding JSONP in the process (or, I could also remove it altogether). It works for now.

**Comment by oooShiny** [comment]
Yep, the official line I've gotten from gfycat support is "...so it looks like doing an update for the JSEmbed is much more involved then we initially thought it would be. Are you able to use our iframe embeds instead?"

**Comment by b4rtt** [comment]
> I've found that new endpoint as well. It does not support JSONP anymore though which is a shame because it's required for this library!

And could you send me example how you did it please?

**Comment by 21stcn** [comment]
Because the API url has changed, now serving JSON, and the loadJSONP function is not working, a quick solution is to fix the API url and then bypass loadJSONP,  using jquery's getJSON instead:

`loadJSONP("https://gfycat.com/cajax/get/" + gfyId, function (data) {`

becomes:

`$.getJSON("https://api.gfycat.com/v1/gfycats/" + gfyId, function(data) {`

Requires jQuery (Replace '$' with 'jQuery' or whatever if you are using jQuery noConflict()).

Disclaimer: This works for my purposes of getting my site's videos back online but is not fully tested in various situation, and does not remove redundant code.

**Comment by igormakovsky** [comment]
guys, is there any further info on proper solution of this problem?

**Comment by igormakovsky** [comment]
the friend of mine have written this piece of code to make gfy videos appear on the page again:

`$( document ).ready(function() {
var url = 'https://api.gfycat.com/v1/gfycats/'
  $('.gfyitem').each(function(index) {
    var gifID = $(this).data('id')
    var blockID = $(this).attr('id')
    var itemImg = $(this)
    $.get( url+gifID, function( data ) {
      var videoString = '<div id="'+blockID+'"><video  loop="true" autoplay="true" muted><source src="'+data.gfyItem.mp4Url+'" type="video/mp4; codecs=\"avc1.42E01E, mp4a.40.2\""><source src="'+data.gfyItem.webmUrl+'" type="video/webm; codecs=\"vp8, vorbis\""></video></div>'
        itemImg.replaceWith( 
          videoString
        )
    });
  })
})`

Check [codepen](https://codepen.io/porcelanosa/pen/PXJBrZ?editors=1011)
